examples/features/torch_batch_process/batch_inference/compare_with_core_api/core_api_inference.py

```diff
@@ -0,0 +1,168 @@
+import json
+import logging
+import math
+import os
+import pathlib
+from typing import Any, Optional
+
+import constants
+import filelock
+import model
+import torch
+import torch.distributed as dist
+import torchvision as tv
+from torch import nn
+from torch.utils import data
+from torchvision import transforms
+
+import determined as det
+from determined import core, pytorch
+
+logging.getLogger().setLevel(logging.INFO)
+
+
+def run_inference(
+    ml_model: nn.Module,
+    data_loader: Any,
+    context: core.Context,
+    rank: int,
+    skip: int,
+    per_worker_iterate_length: int,
+    pred_dir: pathlib.Path,
+    checkpoint_interval: int,
+) -> None:
+    ml_model.eval()
+    dataloader_iterator = iter(data_loader)
+
+    with torch.no_grad():
+        last_checkpoint_step = skip
+        steps_completed = skip
+        for batch_idx in range(skip, per_worker_iterate_length):
+            X = next(dataloader_iterator, None)
+            if X is not None:
+                data, label = X
```

Comment:
does `data, label c next(dataloader_iterator, (None, None))` work?

That would make the code a bit more readable I think.